react/react-0.13.0.d.ts

```diff
@@ -0,0 +1,958 @@
+// Type definitions for React 0.13.0
+// Project: http://facebook.github.io/react/
+// Definitions by: Asana <https://asana.com>, AssureSign <http://www.assuresign.com>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+declare module React {
+    //
+    // React Elements 
+    // ----------------------------------------------------------------------
+    
+    type ReactType = ComponentClass<any, any, any> | string;
+
+    interface ReactElement<P> {
+        type: ReactType;
+        props: P;
+        key: number | string;
+        ref: string;
+    }
+
+    interface ReactHTMLElement extends ReactElement<HTMLAttributes> {}
+    interface ReactSVGElement extends ReactElement<SVGAttributes> {}
+
+    //
+    // React Nodes 
+    // http://facebook.github.io/react/docs/glossary.html
+    // ----------------------------------------------------------------------
+
+    type ReactText = string | number;
+    type ReactChild = ReactElement<any> | ReactText;
+
+    // Should be Array<ReactNode> but type aliases cannot be recursive
+    type ReactFragment = Array<ReactChild | any[] | boolean>;
+    type ReactNode = ReactChild | ReactFragment | boolean;
+
+    //
+    // React Components 
+    // ----------------------------------------------------------------------
+
+    interface ComponentLifecycle<P, S, C> {
+        componentWillMount?(): void;
+        componentDidMount?(): void;
+        componentWillReceiveProps?(nextProps: P, nextContext: C): void;
+        shouldComponentUpdate?(nextProps: P, nextState: S, nextContext: C): boolean;
+        componentWillUpdate?(nextProps: P, nextState: S, nextContext: C): void;
+        componentDidUpdate?(prevProps: P, prevState: S, prevContext: C): void;
+        componentWillUnmount?(): void;
+    }
+    
+    // "modern" ES6 classes
+    class Component<P, S, C> implements ComponentLifecycle<P, S, C> {
+        constructor(props: P, context: C);
+        // these can't be type checked with generics. However, see ComponentClass<P, S, C>
```

Comment:
nvm, got it—can you `s/these/static members` to be more specific? Not sure why the reference to `ComponentClass<P, S, C>` is necessary here either